SCAutolib/models/file.py

```diff
@@ -123,6 +134,50 @@ def set(self, key: str, value: Union[int, str, bool], section: str = None):
             logger.debug(f"Old value in section [{section}] {key}={previous}")
             logger.debug(f"New value in section [{section}] {key}={value}")
 
+    def get(self, key, section: str = None, separator: str = "="):
+        """
+        Method would return the value of the key in section (if set). If the
+        file do not support sections (section for this method is not provided),
+        then file would be parsed in a simple way with `.readlines()` method and
+        using separator each line would be split into two parts: key and a
+        value. If key is matched, the value is striped and returned.
+
+        If file supports section (can be parsed via configparser), then section
+        (in this case has to be set) and key would be used for accessing the
+        value.
+
+        In both cases, if key is not found, an exception would be raised:
+        SCAutolib.SCAutolibException for no-configparser files, and for
+        configparser some of its exceptions.
+        :param key: required key
+        :param section: section where the key should be found
+        :param separator: applicable only for non-configparser file. Separator
+            that would be used to so split a line from the file. By default
+            separator is '='
+
+        :return: value of the key in section (if set)
+        """
+        if section is None:
+            # simple config files without sections
```

Comment:
Support for getting values from section-less config is fragile and due to lack of "logic" in section-less configs this can not be reliably improved. This will return value for first occurrence, however, there can be multiple occurrences of the same key in section-less file. As we discussed earlier, section-less config files are problematic and we support them rather symbolically. There should be at least some warning saying "You are applying get method on non-standard config file. This method would return only the first match". However, this is problematic itself, because some tools may use last match rather than first. IMHO, better approach would be to check if the key is present in the config multiple times and raise an exception if true.

Reply:
Can you please fix this via PR?